### ISSUE TITLE
Can we use the generated "yolov4-custom.cfg" on new training with increased classes ?

### ISSUE THREAD

**Issue by rajaththomson** [AlexeyAB/darknet#5703]
Supposed that, I have 3 classes of objects [cat, dog, fish] and each class contains 450 of labelled images for training.
After finished the training with using pretrained weights-file (such as yolov4.conv.137) and, later that, I want to expand the objects to 6 classes [cat, dog, fish, bird, horse, bear] what should I do?

A. Using the previous trained weights-file (from 3 classes) and training with the image dataset of only additional 3 classes [bird, horse, bear]. The file obj.names are contained 6 classes as [cat, dog, fish, bird, horse, bear].

B. Using the previous trained weights-file (from 3 classes) and training with the image dataset contains all of 6 classes [cat, dog, fish, bird, horse, bear]. The images dataset is mixed between [cat, dog, fish] images from the previous training and the new images for [bird, horse, bear]. The file obj.names are contained 6 classes as [cat, dog, fish, bird, horse, bear].

C. Discard the previous train weights-file and start training from the beginning by using the yolov4.conv.137 weights-file again.

**Comment by AlexeyAB** [comment]
B

**Comment by NHarishGit** [comment]
> B

@AlexeyAB, wouldn't B, over train the model on [cat, dog, fish] ?

**Comment by AlexeyAB** [comment]
* C - better final accuracy

* B - training slightly faster, just do `darknet.exe partial cfg/yolo-obj.cfg yolo-obj_final.weights yolo-obj.conv.137 137` and use `yolo-obj.conv.137` as pre-trained weights file

**Comment by rajaththomson** [comment]
@AlexeyAB I tried with B and this is what happens:

1. The weight generated from training the "3 classes" successfully tests and gives results for cat, dog, fish
2. I use this weight and increase classes to 6 and train again
3. After about 100 iterations for training "6 classes" I stop to get the new weight generated
4. This new weight is now not able to detect even the earlier 3 classes ie:  cat, dog, fish

Shouldn't this  new weight be immediately (even with 100 iterations)  be giving results for earlier 3 classes as it was already trained for it? or is there some mistake on my end with the newly added classes ?

**Comment by AlexeyAB** [comment]
No.

**Comment by rajaththomson** [comment]
@AlexeyAB  So the new weight even with "darknet.exe partial" method , needs to be trained more than 100 iterations to give out results for the first 3 classes?

**Comment by xuemeizh** [comment]
@rajaththomson Any updates so far?

**Comment by stephanecharette** [comment]
@xuemeizh You're asking for an update from someone who posted over 2 years ago with an incorrect usage?
